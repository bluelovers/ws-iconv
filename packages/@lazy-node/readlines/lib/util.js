"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.splitBufferByBuffer = exports.bufferIndexWith = exports.bufferIndexWithByBuffer = exports.bufferStripEndWith = exports.bufferStripEndWithByBuffer = exports.bufferEndWith = exports.bufferEndWithByBuffer = exports.handleOptionNewLineCharacter = void 0;
function handleOptionNewLineCharacter(newLineCharacter) {
    if (!newLineCharacter) {
        return [10 /* EnumNewLineCharacter.LF */];
    }
    else if (typeof newLineCharacter !== 'number') {
        let ls = [];
        for (let i = 0; i < newLineCharacter.length; i++) {
            let c = newLineCharacter[i];
            if (typeof c !== 'number') {
                ls.push(c.charCodeAt(0));
            }
            else {
                ls.push(c);
            }
        }
        return ls;
    }
    return [newLineCharacter];
}
exports.handleOptionNewLineCharacter = handleOptionNewLineCharacter;
function bufferEndWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    return i !== -1 && (i + value.length) === buf.length;
}
exports.bufferEndWithByBuffer = bufferEndWithByBuffer;
function bufferEndWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        return buf[buf.length - 1] === value;
    }
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    return i !== -1 && (i + value.length) === buf.length;
}
exports.bufferEndWith = bufferEndWith;
function bufferStripEndWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    if (i !== -1 && (i + value.length) === buf.length) {
        return buf.slice(0, i);
    }
    return buf;
}
exports.bufferStripEndWithByBuffer = bufferStripEndWithByBuffer;
function bufferStripEndWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        if (buf[buf.length - 1] === value) {
            return buf.slice(0, buf.length - 1);
        }
    }
    else {
        let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
        if (i !== -1 && (i + value.length) === buf.length) {
            /*
            console.dir({
                i,
                vl: value.length,
                bl: buf.length,

                v: value.toString(),
                b: buf.toString(),
                b2: buf.slice(0, i).toString(),
            })
             */
            return buf.slice(0, i);
        }
    }
    return buf;
}
exports.bufferStripEndWith = bufferStripEndWith;
function bufferIndexWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.indexOf(value, byteOffset, encoding);
    return i !== -1 && i === byteOffset;
}
exports.bufferIndexWithByBuffer = bufferIndexWithByBuffer;
function bufferIndexWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        return buf[byteOffset] === value;
    }
    let i = buf.indexOf(value, byteOffset, encoding);
    return i !== -1 && i === byteOffset;
}
exports.bufferIndexWith = bufferIndexWith;
function splitBufferByBuffer(buffer, value) {
    const lines = [];
    let bufferPosition = 0;
    let lastNewLineBufferPosition = 0;
    while (true) {
        let line;
        let bufferPositionValue = buffer[bufferPosition];
        if (bufferIndexWithByBuffer(buffer, value, bufferPosition++)) {
            line = buffer.slice(lastNewLineBufferPosition, bufferPosition += (value.length - 1));
            lines.push(line);
            lastNewLineBufferPosition = bufferPosition;
        }
        else if (bufferPositionValue === undefined) {
            break;
        }
    }
    let leftovers = buffer.slice(lastNewLineBufferPosition, bufferPosition);
    if (leftovers.length) {
        lines.push(leftovers);
    }
    return lines;
}
exports.splitBufferByBuffer = splitBufferByBuffer;
//# sourceMappingURL=util.js.map