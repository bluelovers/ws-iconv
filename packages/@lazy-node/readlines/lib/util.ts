import { INewLineCharacter, EnumNewLineCharacter } from './types';

export function handleOptionNewLineCharacter(newLineCharacter?: INewLineCharacter): number[]
{
	if (!newLineCharacter)
	{
		return [EnumNewLineCharacter.LF];
	}
	else if (typeof newLineCharacter !== 'number')
	{
		let ls = [] as number[]

		for (let i = 0; i < newLineCharacter.length; i++)
		{
			let c = newLineCharacter[i];

			if (typeof c !== 'number')
			{
				ls.push(c.charCodeAt(0))
			}
			else
			{
				ls.push(c)
			}
		}

		return ls
	}

	return [newLineCharacter]
}

export function bufferEndWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	return i !== -1 && (i + value.length) === buf.length
}

export function bufferEndWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	if (typeof value === 'number')
	{
		return buf[buf.length - 1] === value
	}

	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	return i !== -1 && (i + value.length) === buf.length
}

export function bufferStripEndWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	if (i !== -1 && (i + value.length) === buf.length)
	{
		return buf.slice(0, i)
	}

	return buf
}

export function bufferStripEndWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	if (typeof value === 'number')
	{
		if (buf[buf.length - 1] === value)
		{
			return buf.slice(0, buf.length - 1);
		}
	}
	else
	{
		let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

		if (i !== -1 && (i + value.length) === buf.length)
		{
			/*
			console.dir({
				i,
				vl: value.length,
				bl: buf.length,

				v: value.toString(),
				b: buf.toString(),
				b2: buf.slice(0, i).toString(),
			})
			 */

			return buf.slice(0, i)
		}
	}

	return buf
}

export function bufferIndexWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset: number,
	encoding?: BufferEncoding,)
{
	let i = buf.indexOf(value, byteOffset, encoding)

	return i !== -1 && i === byteOffset
}

export function bufferIndexWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset: number,
	encoding?: BufferEncoding,)
{
	if (typeof value === 'number')
	{
		return buf[byteOffset] === value
	}

	let i = buf.indexOf(value, byteOffset, encoding)

	return i !== -1 && i === byteOffset
}

export function splitBufferByBuffer(buffer: Buffer, value: Uint8Array)
{
	const lines: Buffer[] = [];
	let bufferPosition = 0;

	let lastNewLineBufferPosition = 0;
	while (true)
	{
		let line: Buffer;
		let bufferPositionValue = buffer[bufferPosition];

		if (bufferIndexWithByBuffer(buffer, value, bufferPosition++))
		{
			line = buffer.slice(lastNewLineBufferPosition, bufferPosition += (value.length - 1));
			lines.push(line);
			lastNewLineBufferPosition = bufferPosition;
		}
		else if (bufferPositionValue === undefined)
		{
			break;
		}
	}

	let leftovers = buffer.slice(lastNewLineBufferPosition, bufferPosition);
	if (leftovers.length)
	{
		lines.push(leftovers);
	}

	return lines;
}
