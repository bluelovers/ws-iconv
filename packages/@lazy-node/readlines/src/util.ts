import { INewLineCharacter } from './types';
import { EnumNewLineCharacter } from './index';

export function handleOptionNewLineCharacter(newLineCharacter?: INewLineCharacter): number[]
{
	if (!newLineCharacter)
	{
		return [EnumNewLineCharacter.LF];
	}
	else if (typeof newLineCharacter !== 'number')
	{
		const ls = [] as number[];

		for (let i = 0; i < newLineCharacter.length; i++)
		{
			const c = newLineCharacter[i];

			if (typeof c !== 'number')
			{
				ls.push(c.charCodeAt(0))
			}
			else
			{
				ls.push(c)
			}
		}

		return ls
	}

	return [newLineCharacter]
}

