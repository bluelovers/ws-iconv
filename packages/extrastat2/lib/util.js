"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.reassemble = void 0;
const path_1 = require("path");
function reassemble(pathparts) {
    return (0, path_1.join)('/', ...pathparts);
}
exports.reassemble = reassemble;
//# sourceMappingURL=util.js.map