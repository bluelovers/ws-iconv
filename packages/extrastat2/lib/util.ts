import { join } from "path";

export function reassemble(pathparts: string[])
{
	return join('/', ...pathparts)
}
