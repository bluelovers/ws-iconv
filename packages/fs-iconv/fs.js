"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.iconv = exports.saveFile = exports.loadFileSync = exports.loadFile = exports._outputStream = exports._createStreamPassThrough = exports._autoDecode = exports.saveFileSync = exports.ensureWriteStream = void 0;
/**
 * Created by user on 2019/3/17.
 */
const core_1 = require("./core");
const core_2 = require("./core");
Object.defineProperty(exports, "ensureWriteStream", { enumerable: true, get: function () { return core_2.ensureWriteStream; } });
Object.defineProperty(exports, "saveFileSync", { enumerable: true, get: function () { return core_2.saveFileSync; } });
Object.defineProperty(exports, "_autoDecode", { enumerable: true, get: function () { return core_2._autoDecode; } });
Object.defineProperty(exports, "_createStreamPassThrough", { enumerable: true, get: function () { return core_2._createStreamPassThrough; } });
Object.defineProperty(exports, "_outputStream", { enumerable: true, get: function () { return core_2._outputStream; } });
Object.defineProperty(exports, "loadFile", { enumerable: true, get: function () { return core_2.loadFile; } });
Object.defineProperty(exports, "loadFileSync", { enumerable: true, get: function () { return core_2.loadFileSync; } });
Object.defineProperty(exports, "saveFile", { enumerable: true, get: function () { return core_2.saveFile; } });
const iconv_jschardet_1 = __importDefault(require("iconv-jschardet"));
exports.iconv = iconv_jschardet_1.default;
__exportStar(require("fs-extra"), exports);
const fs = core_1.WrapFSIconv(require('fs-extra'));
// @ts-ignore
exports = module.exports = fs;
//# sourceMappingURL=fs.js.map