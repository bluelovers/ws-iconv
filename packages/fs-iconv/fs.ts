/**
 * Created by user on 2019/3/17.
 */
import { WrapFSIconv, SymFSLib } from './core';
import { ensureWriteStream, saveFileSync, _autoDecode, _createStreamPassThrough, _outputStream, loadFile, loadFileSync, saveFile } from './core';
import { vEncoding } from 'iconv-jschardet';
import fsExtra = require('fs-extra');
import clone = require("lodash/clone");
import iconv from 'iconv-jschardet';
import Bluebird = require('bluebird');
import stream = require('stream');

export * from 'fs-extra';

export { ensureWriteStream, saveFileSync, _autoDecode, _createStreamPassThrough, _outputStream, loadFile, loadFileSync, saveFile }

export import IWrapFSIconvOptions = WrapFSIconv.IWrapFSIconvOptions;
export import IWrapFSIconvOptionsLoadFile = WrapFSIconv.IWrapFSIconvOptionsLoadFile;
export import IWrapFSIconvOptionsLoadFile2 = WrapFSIconv.IWrapFSIconvOptionsLoadFile2;
export import IEncoding = WrapFSIconv.IEncoding;


export { iconv }

const fs = WrapFSIconv(require('fs-extra') as typeof import('fs-extra'));

// @ts-ignore
exports = module.exports = fs;
