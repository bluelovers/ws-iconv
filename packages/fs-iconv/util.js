"use strict";
/**
 * Created by user on 2019/3/17.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.trimFilename = void 0;
const sanitize_filename_1 = __importDefault(require("sanitize-filename"));
function trimFilename(name) {
    let ret = name.toString()
        .replace(/\r\n|\r|\n/g, ' ')
        .replace(/[\r\n\t  \xA0]+/g, ' ');
    return sanitize_filename_1.default(ret)
        .trim()
        .replace(/^[　\s_]+/g, '')
        .replace(/[　\s_]+$/g, '');
}
exports.trimFilename = trimFilename;
exports.default = exports;
//# sourceMappingURL=util.js.map