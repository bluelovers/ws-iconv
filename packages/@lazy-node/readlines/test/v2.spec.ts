import LineByLine from '../class';
import { join } from 'path';
import { outputFileSync } from 'fs-extra';
import { openSync } from "fs";
import { newContent } from './util';

describe('v2', () =>
{

	let file = join(__dirname, './temp/', `tmp.txt`);

	let newLineCharacter = `\r\n`;
	let totalLines = 10;

	let content = newContent(newLineCharacter, totalLines);

	outputFileSync(file, content);

	test(`content`, () =>
	{
		expect(content).toMatchSnapshot();
	})

	test(`prop`, () =>
	{
		const liner = new LineByLine(file, {
			newLineCharacter,
		});

		expect(liner).toHaveProperty('file')
		expect(liner).toHaveProperty('fd')
		expect(liner).toHaveProperty('options')
		expect(liner).toHaveProperty('fdPosition')
		expect(liner).toHaveProperty('newLineCharacter')
		expect(liner).toHaveProperty('eofReached')
		expect(liner).toHaveProperty('lineNumber')

		liner.close();
	})

	test(`fd`, () =>
	{

		let fd = openSync(file, 'r');

		const liner = new LineByLine(fd, {
			newLineCharacter,
		});

		expect(liner).toHaveProperty('fd', fd);
		expect(liner).toHaveProperty('file', fd);

		liner.close();
	})

	test(`lineNumber`, () =>
	{
		const liner = new LineByLine(file, {
			newLineCharacter,
		});

		let line: Buffer;
		let lineNumber = 0;

		expect(liner.lineNumber).toStrictEqual(-1);
		expect(liner.lineNumber).toMatchSnapshot();

		while (line = liner.next())
		{
			expect(liner.lineNumber).toStrictEqual(lineNumber);
			expect(liner.lineNumber).toMatchSnapshot();

			lineNumber++;
		}

		expect(liner.lineNumber).toStrictEqual(totalLines - 1)
		expect(liner.lineNumber).toMatchSnapshot();

	});

	test(`string`, () =>
	{
		const liner = new LineByLine(file, {
			newLineCharacter,
		});

		let line: Buffer;

		while (line = liner.next())
		{
			let s = line.toString();
			expect(s).not.toMatch(/\r|\n/);
			expect(s).toMatchSnapshot();
		}

	});

	test(`generator`, () =>
	{
		const liner = new LineByLine(file, {
			newLineCharacter,
		});

		let line: Buffer;

		for (line of liner.generator())
		{
			let s = line.toString();
			expect(s).not.toMatch(/\r|\n/);
			expect(s).toMatchSnapshot();
		}

		expect(liner.lineNumber).toStrictEqual(totalLines - 1)

	});

	test(`generator static`, () =>
	{
		let line: Buffer;
		let lineNumber = 0;

		const liner = LineByLine.generator(file, {
			newLineCharacter,
		})

		for (line of liner)
		{
			let s = line.toString();
			expect(s).not.toMatch(/\r|\n/);
			expect(s).toMatchSnapshot();

			lineNumber++;
		}

		expect(lineNumber).toStrictEqual(totalLines)

	});

	test(`generator static v2`, () =>
	{
		let line: Buffer;
		let lineNumber = 0;

		for (line of LineByLine.generator(file, {
			newLineCharacter,
		}))
		{
			let s = line.toString();
			expect(s).not.toMatch(/\r|\n/);
			expect(s).toMatchSnapshot();

			lineNumber++;
		}

		expect(lineNumber).toStrictEqual(totalLines)

	});

})

