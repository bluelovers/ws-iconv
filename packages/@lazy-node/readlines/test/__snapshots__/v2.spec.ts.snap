// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v2 content 1`] = `
"0
1
2
3
4
5
6
7
8
9"
`;

exports[`v2 generator 1`] = `"0"`;

exports[`v2 generator 2`] = `"1"`;

exports[`v2 generator 3`] = `"2"`;

exports[`v2 generator 4`] = `"3"`;

exports[`v2 generator 5`] = `"4"`;

exports[`v2 generator 6`] = `"5"`;

exports[`v2 generator 7`] = `"6"`;

exports[`v2 generator 8`] = `"7"`;

exports[`v2 generator 9`] = `"8"`;

exports[`v2 generator 10`] = `"9"`;

exports[`v2 generator static 1`] = `"0"`;

exports[`v2 generator static 2`] = `"1"`;

exports[`v2 generator static 3`] = `"2"`;

exports[`v2 generator static 4`] = `"3"`;

exports[`v2 generator static 5`] = `"4"`;

exports[`v2 generator static 6`] = `"5"`;

exports[`v2 generator static 7`] = `"6"`;

exports[`v2 generator static 8`] = `"7"`;

exports[`v2 generator static 9`] = `"8"`;

exports[`v2 generator static 10`] = `"9"`;

exports[`v2 generator static v2 1`] = `"0"`;

exports[`v2 generator static v2 2`] = `"1"`;

exports[`v2 generator static v2 3`] = `"2"`;

exports[`v2 generator static v2 4`] = `"3"`;

exports[`v2 generator static v2 5`] = `"4"`;

exports[`v2 generator static v2 6`] = `"5"`;

exports[`v2 generator static v2 7`] = `"6"`;

exports[`v2 generator static v2 8`] = `"7"`;

exports[`v2 generator static v2 9`] = `"8"`;

exports[`v2 generator static v2 10`] = `"9"`;

exports[`v2 lineNumber 1`] = `-1`;

exports[`v2 lineNumber 2`] = `0`;

exports[`v2 lineNumber 3`] = `1`;

exports[`v2 lineNumber 4`] = `2`;

exports[`v2 lineNumber 5`] = `3`;

exports[`v2 lineNumber 6`] = `4`;

exports[`v2 lineNumber 7`] = `5`;

exports[`v2 lineNumber 8`] = `6`;

exports[`v2 lineNumber 9`] = `7`;

exports[`v2 lineNumber 10`] = `8`;

exports[`v2 lineNumber 11`] = `9`;

exports[`v2 lineNumber 12`] = `9`;

exports[`v2 string 1`] = `"0"`;

exports[`v2 string 2`] = `"1"`;

exports[`v2 string 3`] = `"2"`;

exports[`v2 string 4`] = `"3"`;

exports[`v2 string 5`] = `"4"`;

exports[`v2 string 6`] = `"5"`;

exports[`v2 string 7`] = `"6"`;

exports[`v2 string 8`] = `"7"`;

exports[`v2 string 9`] = `"8"`;

exports[`v2 string 10`] = `"9"`;
